api-put-object-streaming.go

```diff
@@ -419,6 +419,7 @@ func (c *Client) putObject(ctx context.Context, bucketName, objectName string, r
 		return UploadInfo{}, errInvalidArgument("MD5Sum cannot be calculated with size '-1'")
 	}
 
+	var readSeeker io.Seeker
```

Comment:
This gets set when `reader` is `io.ReaderAt` *and* `io.Seeker`.

It seems like it should also be possible with it just being `io.Seeker` (but still excluding stdin/out).

Maybe we can add that in the future.

Reply:
No it will only get set for the `io.SectionReader`